frontend/src/features/form/utils/augmentWithWorkflowDisabling.ts

```diff
@@ -0,0 +1,23 @@
+import { FormFieldDto, FormWorkflowStep } from '~shared/types'
+
+import { NON_RESPONSE_FIELD_SET } from '../constants'
+
+export const isFieldEnabledByWorkflow = (
+  workflowStep: FormWorkflowStep | undefined,
+  field: FormFieldDto,
+) =>
+  // Field is enabled if:
+  // 1. there is no workflow OR
+  // 2. it is a non-response field OR
+  // 3. (by this point it is workflow + response field) has been explicitly set to be editable in the workflow
+  !workflowStep ||
+  NON_RESPONSE_FIELD_SET.has(field.fieldType) ||
+  workflowStep.edit.includes(field._id)
```

Comment:
/nit Wondering if this would be clearer, given that this logic can be tricky (i.e., needing expressively requiring comments to provide context).

```suggestion
) => {
  // Field is enabled if there is no workflow
  if (!workflowStep) return true
  // Field is enabled if it is a non-response field
  if (NON_RESPONSE_FIELD_SET.has(field.fieldType)) return true
  // (by this point it is workflow + response field) has been explicitly set to be editable in the workflow
  return workflowStep.edit.includes(field._id)
}
```

Reply:
point taken, updated in [296d3a4](https://github.com/opengovsg/FormSG/pull/7091/commits/296d3a41e92c683fd4fde1749fdd3c1cff3eef1f)